<a href="https://colab.research.google.com/github/asaputro281-boo/All-Project/blob/main/P01_253307019_Angga_Dwi_Saputro.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Kuiz**

*Bab 1 Pengantar Sistem Cerdas*

Pilihan Ganda
1. Ciri yang paling membedakan sistem cerdas dari sistem konvensional adalah  
Jawaban : (c) mampu mengambil keputusan masuk akal pada situasi yang tidak sepenuhnya terprogram  
Pembahasan : Pembeda paling mendasar antara sistem cerdas sama sistem konvensional itu ada di fleksibilitasnya. Sistem cerdas tetap bisa ngasih keputusan yang masuk akal pas nemu input atau kasus baru yang belum pernah di-koding sama sekali.
2. Uji Turing Menilai kecerdasan mesin berdasarkan  
jawaban : (b) ketakterbedaan jawabannya dari jawaban manusia oleh penanya  
Pembahasan : Alan Turing (1950) bikin standar simpel buat ngukur kecerdasan mesin. Intinya, kalau ada orang yang ngobrol lewat teks dan dia enggak bisa bedain mana jawaban mesin mana jawaban manusia, berarti mesin itu udah dianggap cerdas.
3. Peristiwa yang memicu AI winter pertama adalah   
jawaban : (b) kritik atas keterbatasan perceptron  
Pembahasan : AI Winter* gelombang pertama (1969–1980) terjadi gara-gara kritik Minsky dan Papert terhadap *Perceptron*. Mereka ngebuktiin kalau *Perceptron* tunggal tuh terbatas banget dan enggak bisa nyelesaiin masalah logika sederhana kayak masalah XOR. Efeknya, dana riset AI langsung drop.
4. Dalam kerangka PEAS untuk mobil otonom, kamera dan radar termasuk komponen  
jawaban : (d) sensors  
Pembahasan : Dalam konsep PEAS (*Performance, Environment, Actuators, Sensors*), kamera sama radar fungsinya buat ngamatin dan nangkap data dari kondisi sekitar. Jadi, dua komponen itu masuknya ke kategori Sensors.
5. Masalah “menyusun jadwal ujian 60 kelas tanpa bentrok dengan ruang terbatas” paling alami diselesaikan rumpun  
jawaban : (c) evolusioner  
Pembahasan : Masalah ngatur jadwal kuliah/ujian biar enggak bentrok itu termasuk masalah **optimasi kombinasi yang sangat rumit**. Rumpun **Evolusioner** (kayak Algoritma Genetika) paling cocok buat nyelesaiin kasus kayak gini karena dia bisa nyari kombinasi jawaban paling pas tanpa harus nyari rumus matematis langsung.


Esai

Jelaskan dengan kata - katamu sendiri tiga kunci definisi sistem cerdas (menalar, belajar, adapatasi) dan beri satu contoh aplikasi untuk masing - masing.   
Jawaban :
1. Menalar (*Reasoning*) : Membuat keputusan atau penarikan kesimpulan    berdasarkan aturan/pengetahuan yang udah tersimpan di sistem.
**Contoh :** *Sistem Pakar Diagnosa Laptop*. Sistem bakal nanya gejala-gejalanya dulu (misal: "indikator nyala tapi layar gelap"), lalu nyatuin aturan yang ada buat nyimpulkan kalau RAM-nya yang bermasalah.
2. Belajar (*Learning*) : Kemampuan sistem buat nemuin pola sendiri dari sekumpulan data historis, tanpa harus dituliskan aturannya satu per satu secara manua **Contoh :** *Filter Spam Email*. Sistem bakal nyelipin email ke folder spam setelah mempelajari pola kata-kata dari ribuan email spam yang pernah ada sebelumnya.
3. Beradaptasi (*Adapting*) : Penyesuaian respons/tindakan sistem secara otomatis pas nemuin perubahan kondisi lingkungan atau kebiasaan baru.
**Contoh :** *Aplikasi Navigasi (Google Maps)*. Pas jalanan yang kita lewati mendadak macet total, aplikasinya langsung otomatis nyariin rute alternatif lain.


Mengapa deep learning baru meledak pada 2012 padahal backpropagation sudah ditemukan 1986 ? kaitkan dengan tiga syarat lompatan AI   

Jawaban :
1. Ketersediaan Algoritma: Konsep dasar pelatihan jaringan saraf kayak *backpropagation* emang udah ada dari tahun 1986. Tapi kalau cuma modal algoritma doang tanpa didukung dua hal lainnya, hasilnya belum bisa maksimal.
2. **Data yang Melimpah:** Di era 80-an, data digital masih sedikit banget. Pas internet udah makin berkembang pesat (terutama pas ada *dataset* raksasa kayak ImageNet), barulah AI punya pasokan bahan latih yang cukup biar modelnya makin pinter dan enggak gampang salah (*overfitting*).
3. **Daya Komputasi (*GPU*):** Mengolah *deep learning* itu butuh perhitungan matriks yang sangat masif. Nah, baru di sekitaran tahun 2012-an para peneliti (seperti tim AlexNet) mulai manfaatin kartu grafis GPU buat mroses data latih ribuan kali lebih cepat ketimbang cuma ngandelin CPU biasa.


Susun deskripsi PEAS untuk sistem rekomendasi kantin kampus (menu apa yang ditawarkan ke tiap mahasiswa)  

Jawaban :
Deskripsi PEAS untuk Sistem Rekomendasi Kantin Kampus:

1. Performance Measure (Ukuran Sukses):
   - Berapa banyak mahasiswa yang beneran ngebeli menu hasil rekomendasi.
   - Tingkat kepuasan atau rating dari mahasiswa setelah makan.
   - Variasi rekomendasi (menu enggak itu-itu aja tiap hari).
   - Seberapa cepat mahasiswa nemuin menu yang disukai.

2. Environment (Lingkungan):
   - Stok menu kantin yang tersedia secara real-time.
   - Profil mahasiswa (budget, alergi makanan, riwayat pesanan).
   - Jam makan (pagi, siang, atau sore).
   - Cuaca (kalau hujan, sistem cenderung nawarin makanan berkuah/hangat).

3. Actuators (Tindakan/Output):
   - Daftar pilihan menu rekomendasi yang muncul di aplikasi kantin.
   - Notifikasi promo menu harian yang dikirim ke HP mahasiswa.

4. Sensors (Input Data):
   - Riwayat pencarian dan transaksi makanan sebelumnya.
   - Input kriteria dari mahasiswa (misal: budget maksimal, opsi vegetarian).
   - Jam pemesanan dan data cuaca lokal saat itu.

  
Berikan satu contoh sistem disekitarmu yang mengggabungkan lebih dari satu rumpun pendekatan , dan uraikan pembagian perannya.

Jawaban :
**Contoh Sistem:** Platform Aplikasi E-Commerce (seperti Tokopedia / Shopee)

1. **Rumpun Statistik / Pembelajaran (*Machine Learning / Deep Learning*):**
   * **Perannya:** Dipakai untuk **Sistem Rekomendasi Produk** dan **Pencarian Berbasis Gambar (Image Search)**. Algoritma mempelajari pola perilaku/riwayat transaksi jutaan pembeli untuk merekomendasikan barang yang paling relevan di beranda, serta mengenali objek foto baju/sepatu yang diunggah pengguna untuk mencari produk serupa.

2. **Rumpun Simbolik (*Pencarian & Sistem Berbasis Aturan*):**
   * **Perannya:** Dipakai untuk **Aturan Transaksi (Promo/Voucher)** dan **Algoritma Pencarian Rute Pengiriman**.
     * Aturan logika eksplisit (`IF-THEN`) digunakan untuk memvalidasi syarat kupon promo (misal: `IF` total belanja > Rp100.000 `AND` metode bayar = GoPay `THEN` potongan ongkir = Rp10.000).
     * Algoritma pencarian digunakan oleh sistem logistik kurir untuk merencanakan rute pengiriman paket terpendek dari gudang ke rumah pembeli.

3. **Rumpun Evolusioner (*Algoritma Genetika*):**
   * **Perannya:** Dipakai untuk **Optimasi Penataan Stok & Alokasi Logistik Gudang (Warehouse Management)**. Algoritma genetika mengevolusikan dan menemukan kombinasi tata letak barang terbaik di gudang raksasa agar jarak tempuh staf/robot penjemput barang (*picker*) menjadi paling efisien.


Semakin banyak aturan IF - THEN ditulis , semakin cerdas sistemnya. Setujukah kamu ? Bahas dengan merujuk pelajaran era sistem pakar dan kelahir machine learning

Jawaban :
**SANGAT TIDAK SETUJU.**

1. **Pelajaran dari Era Sistem Pakar (1980-an):**
   * Dulu orang-orang mikir kalau kita nambahin puluhan ribu aturan `IF-THEN` buatan pakar, sistemnya bakal otomatis makin pintar.
   * **Kenyataannya:** Pas aturan udah kebanyakan, sistem malah makin ribet. Banyak aturan yang saling bentrok, biaya perawatannya mahal banget, dan pas nemu kasus baru sedikit aja di luar aturan, sistemnya langsung "mogok" (*brittle*). Masalah ini yang bikin AI sempat memasuki *AI Winter* kedua.

2. **Kelahiran Machine Learning (1990-an–Sekarang):**
   * Karena keterbatasan itu, pendekatan AI akhirnya bergeser. Daripada cape-cape nulis aturan satu-satu secara manual, mending **biarkan mesin yang belajar sendiri dari pola data**.
   * Kasus rumit kayak ngenalin wajah, deteksi suara, atau ngolah teks bebas itu sifatnya fleksibel banget, jadi hampir enggak mungkin kalau cuma diakalin pakai deretan aturan `IF-THEN` buatan manusia.

#**Modul 01 Kontrak Kuliah Peta Sistem Cerdas , Instalasi Jupyter**   
#**Kegiatan 4 - Program Python Pertama**

#**LANGKAH KESATU**

In [ ]:
# Langkah 1
# Program Pertama saya dimata kuliah Sistem Cerdas #(# merupakan komentar gunanya untuk manusia yang membaca kode)
print ("Halo , sistem cerdas") #print merupakan sebuah eksekusi untuk mencetak apa yang ada didalam tanda kurung
print ("Nama Saya Angga Dwi Saputro")

Halo , sistem cerdas
Nama Saya Angga Dwi Saputro


# **LANGKAH KEDUA**





In [ ]:
# Langkah 2
# Python bisa langsung dipakai berhitung
# Variabel adalah kotak penyimpanan nilai yang diberi nama; tanda = berisi isi dengan bukan sama dengan
jp_teori = 3  # jumlah JP teori , Variabel nya bernama jp_teori dengan isi angka 3
jp_praktik = 5  # jumlah JP praktikn , Variabel kedua jp_praktik dengan isi angka 5
menit_per_jp = 50  # 1 JP = 50 menit , variabel ketiga menit_per_jp yang berisi 50 menit untuk satu jp

# Menhitung total menit
total_menit = (jp_teori + jp_praktik) * menit_per_jp # untuk beris code #total menit ini merupakan rumus yang nanti yang ada didalam kurung didahulukan baru nanti dikali
print("Total menit kuliah hari ini:", total_menit)# kemudian baris yang terakhir print ini untuk melakukan eksekusi mencetak hasil akhir

Total menit kuliah hari ini: 400


#**LANGKAH KETIGA**



#Notebook Pertemuan 1
Nama : Angga Dwi Saputro  
NIM  : 253307019

###**Catatan :** hari ini saya berhasil menggunakan dasar bahasa python dalam google collabs.

# **Kegiatan 5 — ✍Latihan Mandiri**

# **LATIHAN 1 KEGIATAN 5**
Data Diri
* **Nama :** Angga Dwi Saputro
* **NIM :** 253307019
* **Asal Sekolah :** Sman 1 Kota Madiun
* **Alasan Memilih Prodi Teknologi Informasi :** Ingin Menguasai dan belajar Information Technology



# **LATIHAN 2 KEGIATAN 5**

In [ ]:
# Mencetak 3 teknologi cerdas yang sering digunakan sehari-hari
print("1. Google")
print("2. Instagram")
print("3. Gemini")

1. Google
2. Instagram
3. Gemini


Penjelasan
* print(): merupakan sebuah perintah eksekusi untuk mencetak program
* Tanda Kurung () kegunaannya untuk menampung argumen
* Tanda Kutip "" menandakan bahwa nilai dalamnya adalah tipe data teks atau string  

# **LATIHAN 3 KEGIATAN 5**

In [ ]:
# Kalkulator JP Semester
# Deklarasi variabel sesuai petunjuk modul
jp_per_pertemuan = 8
pertemuan_tatap_muka = 13
menit_per_jp = 50

# Perhitungan total JP dan total menit satu semester
total_jp_semester = jp_per_pertemuan * pertemuan_tatap_muka
total_menit_semester = total_jp_semester * menit_per_jp

# Mencetak hasil
print("Total JP satu semester    :", total_jp_semester, "JP")
print("Total menit satu semester :", total_menit_semester, "menit")

Total JP satu semester    : 104 JP
Total menit satu semester : 5200 menit


Penjelasan Kalkulator JP Semester

*   jp_per_pertemuan = 8 menyimpan angka 8 (jumlah jam perlajaran dalam 1 kali tatap muka )
*   pertemuan_tatap_muka = 13 menyimpan jumlah total pertemuan tatap muka dalam 1 semester
*   menit_per_jp = 50 menyimpan angka 50 untuk durasi per 1JP
*   total_jp_semester: Mengalikan 8 x 13, lalu hasilnya (104) disimpan ke dalam variabel total_jp_semester.
*  total_menit_semester : mengalikan hasil total JP 104 dengan menit 50 yaitu 104 x 50 yang hasilnya 5200
* tanda koma , digunakan untuk menggabungkan teks string dalam tanda kutip dengan nilai dari variabel
*   print("Total JP satu semester    :", total_jp_semester, "JP")
Menampilkan teks deskripsi beserta nilai dari variabel total_jp_semester dan teks "JP"
*  print("Total menit satu semester :", total_menit_semester, "menit")
Menampilkan teks deskripsi beserta nilai dari variabel total_menit_semester dan teks "menit".






